src/events/events.ts

```diff
@@ -368,6 +458,13 @@ export const onPresenceUpdate = async (client: Client, before: Presence, after:
     }
 
     const guild: Guild = after.guild;
+
+    // Force fetch members that aren't cached, to avoid partial data
+    // The joined timestamp is 0 for partial member objects
+    if (!guild.members.cache.get(before.userId).joinedTimestamp) {
+        await guild.members.fetch({ user: before.user, force: true });
+    }
```

Comment:
This section has been added to the presence update handler to avoid having partial user data.